### ISSUE TITLE
Fixed path to node binary

### ISSUE THREAD

**Issue by henrik242** [node-gradle/gradle-node-plugin#55]
If I configure node like this: 
```
configure<NodeExtension> {
  version = "12.13.0"
  download = true
  workDir = file("${rootDir}/build/nodejs")
}
```
 .. the node binary will end up in `build/nodejs/node-v12.13.0-darwin-x64/bin/node`, which is dependent on both OS and version.

Is it possible to have it end up in a predetermined folder, like `build/nodejs/foo/bin/node`?

**Comment by deepy** [comment]
It'd probably be pretty easy to add a finalizer for `nodeSetup` which does this locally, but I guess your intended use-case is to add it to the path for easier local development?

**Comment by henrik242** [comment]
@deepy Yeah, the use-case is to add it to the path for easier local development and CI builds

**Comment by deepy** [comment]
If you're ok with some extra space taken you could try creating a gradle `Copy` and tell `nodeSetup` that it's [finalizedBy](https://docs.gradle.org/current/dsl/org.gradle.api.Task.html#org.gradle.api.Task:finalizedBy) that task, which would just move it one step up.

But I guess this is something we do want to fix.

**Comment by henrik242** [comment]
@deepy Space is not a problem.  Do we have a property pointing to the folder, or do I have to use a `File` handle to search for it?

**Comment by deepy** [comment]
Something like this, it'd make sure that there's always a `node` folder with the content of `nodeSetup`, all the tasks would still use the `build/nodejs/version/bin/node` but you'd have an easy alias for developers

```groovy
def fixed = tasks.register("fixed", Copy) {
    from nodeSetup
    into 'node'
}
tasks.named("nodeSetup").configure { finalizedBy fixed }
```

**Comment by ryparker** [comment]
I was able to reference the exact path to the node executable in my `build.gradle` using
`"$node.variant.nodeExec"`

I'm using the node-gradle-plugin version `2.2.4`

I hope that helps :)

**Comment by vibpe** [comment]
Is it possible to access this path to the node executable with node-gradle-plugin version 3.0.1
` "$node.variant.nodeExec"` does not work anymore

**Comment by espen42** [comment]
This (dead?) issue is the closest I've found see to my problem:

I'm using nvm to manage node versions locally. And this plugin detects a different version than what's set with `nvm use` (and that the terminal and all other processes are using). As a result, when I'm using `npm link`, the symlinks end up in a location different from what this plugin ends up using. I presume this is some local hiccup on my system that causes non-typical behaviour, but I can't figure it out so far.

The workaround above would likely not work for meg, because of the build sequence and when the symlinks appear (and the fact that `npm link` in a second project wouldn't be able to access a locally installed node in the first project, without adding too much acrobatics in the dev-builds).

If it's not possible to enforce a node workdir, could you please tell me how (or where in the source code) the workdir/nodeversion is detected (or configured)? Then I could probably use that to loosen up whatever it is that locks me to the wrong version.

**Comment by deepy** [comment]
The way that this plugin picks binaries is the same way that most things do, it gets the first one on `PATH`

If you're starting your build from command-line it should get the same npm as you get, I guess a simple way to figure out what's going on is by creating a Gradle task that prints `PATH` and see if it matches your shell.

This issue isn't dead and while it is high on the wishlist it's also a little trickier to solve

**Comment by deepy** [comment]
Can't link to the code as I'm on phone, but if you look at my last PR, the one about improving BSD support it's affecting the relevant places

**Comment by espen42** [comment]
Good tip - and fast! 👍 Thanks, I'll try to see if some inner path is somehow different from the system/terminal path (if it is, any hint on where that might stem from? On macOS 11.5...)

**Comment by sambernet** [comment]
Technically our goal (see comment above) was not to force node binary into a fixed folder (as the original author of this issue requested), but to get the path to the actual binary that is used according to the extension configuration (especially when using `download = true`).

This is now possible again since `3.2.1` when #213 was resolved.
The approach suggested there works fine for our case.
